### PR TITLE
Disable RPC profiling for kineto profilers

### PR DESCRIPTION
**Pull request by rohan-varma**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #76234

RPC profiling is only enabled when the profiler is of legacy type.

Differential Revision: [D35484579](https://our.internmc.facebook.com/intern/diff/D35484579/)

### CODE REVIEW

File: torch/testing/_internal/distributed/rpc/rpc_test.py [hunk 1]
```diff
@@ -2186,6 +2206,11 @@ def _run_test_profiler_with_async_rpc_udf(self):
...
         self._profiler_test_with_rpc(RPCExecMode.ASYNC, my_sleep_func, args=(1,),
                                      use_record_function=True)
+        # Test to ensure that kineto profiler enabled in RPC does not enable
+        # RPC profiling (it is unsupported) and does not result in issues.
+        self._profiler_test_with_rpc(
+            RPCExecMode.ASYNC, my_sleep_func, args=(1,), kineto_profile=True
+        )
 
     @dist_init
     def test_profiler_with_async_rpc_udf(self):
```
Comment (H-Huang) [inline torch/testing/_internal/distributed/rpc/rpc_test.py:2216]: Unrelated, curious about this test. Why does this and the test below it call the same func?

**Review by H-Huang — APPROVED** [review]
LGTM, stamp to unblock

File: torch/testing/_internal/distributed/rpc/rpc_test.py (hunk 1 above)
Comment (rohan-varma) [reply torch/testing/_internal/distributed/rpc/rpc_test.py:2216]: Hmm the single threaded one was supposed to test RPC with num_worker_threads=1. Perhaps that went away in the tensorpipe refactor or somehow?

**Pull request closed without merging**